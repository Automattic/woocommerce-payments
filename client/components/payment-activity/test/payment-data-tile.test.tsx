/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

/**
 * Internal dependencies
 */
import PaymentDataTile from '../payment-data-tile';

declare const global: {
	wcpaySettings: {
		accountDefaultCurrency: string;
		zeroDecimalCurrencies: string[];
		currencyData: Record< string, any >;
		connect: {
			country: string;
		};
	};
};

describe( 'PaymentDataTile', () => {
	global.wcpaySettings = {
		accountDefaultCurrency: 'USD',
		zeroDecimalCurrencies: [],
		connect: {
			country: 'US',
		},
		currencyData: {
			US: {
				code: 'USD',
				symbol: '$',
				symbolPosition: 'left',
				thousandSeparator: ',',
				decimalSeparator: '.',
				precision: 2,
			},
		},
	};

	test( 'renders correctly', () => {
		const { container } = render(
			<PaymentDataTile
				id="total-payment"
				currencyCode="USD"
				label="Total payment volume"
			/>
		);
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders label correctly', () => {
		const label = 'Total payment volume';
		render(
			<PaymentDataTile
				id="total-payment"
				currencyCode="USD"
				label={ label }
				amount={ 123 }
			/>
		);
		expect( screen.getByText( label ) ).toBeInTheDocument();
		expect( screen.getByLabelText( label ) ).toHaveTextContent( '$1.23' );
	} );

	test( 'renders amount correctly', () => {
		const amount = 10000;
		const currencyCode = 'USD';
		render(
			<PaymentDataTile
				id="charges-test-tile"
				label="Charges"
				amount={ amount }
				currencyCode={ currencyCode }
			/>
		);
		const amountElement = screen.getByText( '$100.00' );
		expect( amountElement ).toBeInTheDocument();
	} );

	test( 'renders report link correctly', () => {
		const reportLink = 'https://example.com/report';
		render(
			<PaymentDataTile
				id="charges-test-tile"
				label="Charges"
				amount={ 10000 }
				currencyCode="USD"
				reportLink={ reportLink }
			/>
		);
		const reportLinkElement = screen.getByRole( 'link', {
			name: 'View report',
		} );
		expect( reportLinkElement ).toBeInTheDocument();
		expect( reportLinkElement ).toHaveAttribute( 'href', reportLink );
	} );
} );
