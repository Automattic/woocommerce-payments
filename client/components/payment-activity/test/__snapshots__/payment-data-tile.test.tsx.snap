// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDataTile renders correctly 1`] = `
<div>
  <div
    class="wcpay-payment-data-highlights__item"
  >
    <p
      class="wcpay-payment-data-highlights__item__label"
    >
      <span
        id="total-payment"
      >
        Total payment volume
      </span>
    </p>
    <div
      class="wcpay-payment-data-highlights__item__wrapper"
    >
      <p
        aria-labelledby="total-payment"
        class="wcpay-payment-data-highlights__item__wrapper__amount"
      >
        $0.00
      </p>
    </div>
  </div>
</div>
`;
